### PR TITLE
[AIRFLOW-453] Add XCom Admin page

### PR DESCRIPTION
**Pull request by msumit**
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- _https://issues.apache.org/jira/browse/AIRFLOW-453_

Thanks,
Sumit

### CODE REVIEW

File: airflow/www/app.py [hunk 1]
```diff
@@ -88,6 +88,8 @@ def create_app(config=None):
             models.Connection, Session, name="Connections", category="Admin"))
         av(vs.VariableView(
             models.Variable, Session, name="Variables", category="Admin"))
+        av(vs.XComView(
```
Comment (bolkedebruin) [inline airflow/www/app.py:92]: Please make sure this is not viewable from a normal user's perspective as this can expose sensitive information

**Comment by bolkedebruin** [comment]
Nice. Can you update your commit with a description and proper title?

File: airflow/www/app.py (hunk 1 above)
Comment (msumit) [reply airflow/www/app.py:92]: @bolkedebruin can u plz guide me how to achieve that.. as we don't use any auth  mechanism, I thought adding that into `Admin` category should do the trick.

**Comment by r39132** [comment]
@msumit Pls provide screen shots in this PR. Also, alter the commit subject to something like "Add XCom Admin Page"

**Comment by r39132** [comment]
+1

**Comment by r39132** [comment]
I ran your PR locally. Here's a screenshot. Also, for all UI changes, screenshot are required! 

screenshot 2016-10-01 01 22 48

I can understand the need for the list view and delete operation, but not for the create operation. Why would anyone create an xcom object and isn't that both dangerous and undesirable. I realize that flask makes it very easy to add functionality to views that are not intended. A great example is the create view for dag runs, which makes absolutely no sense to me -- I think @normster might have removed it.. (hopefully) ..

Anyway, consider disabling create. Also, this needs tests.

**Comment by msumit** [comment]
@r39132 Thanks for review.. will remove `delete` option and update the commit msg and PR title as well.. However Im not sure how to add tests for this.. Do we've any UI test coverage as of now?

**Comment by r39132** [comment]
@msumit tick-tock.. please update and ping me. If no movement, I will need to close.

**Comment by msumit** [comment]
@r39132 I can update the PR, but now I'm stuck on 2 things: first is @bolkedebruin suggestion to hide from normal users, which I believe should be happening right now as well.. and second is about UI test coverage, something which is missing for the whole project..

**Comment by r39132** [comment]
The basic test is that the endpoint loads, which we have for all endpoints. We don't have a UI test framework, so that shouldn't be a requirement -- though we do need someone to take ownership of that need in the future. Regarding the "hide from normal users", the entire app is flask-admin. My understanding is that we can enable authentication, but not authorization, because everything is behind the admin URI. So, you should be able to add the @login_required decorator and test with an authentication backend (e.g. ldap). I haven't set that up since Airbnb and Agari both authenticate at an apache reverse proxy, not in Airflow. 

Try sending an email to the dev list to help test your feature for this use case and also ask for help is setting up an auth backend to ensure only authenticated users can see it. 

For any contributor to test, they just need to run `dev/airflowpr work_local`. This will set up your PR to test and it's a snap. 

cc/ @bolkedebruin

**Comment by r39132** [comment]
@msumit I'm starting to question the need to edit XCom. What's the need?

**Comment by msumit** [comment]
@r39132 yeah, only view and search is the main goal here..

**Comment by r39132** [comment]
@msumit you do need to add a basic test here..and then squash your commits.  In tests/core.py, you'll find some basic endpoint testing ... e.g. `test_dag_views(..)`.

**Comment by r39132** [comment]
@msumit This PR is dragging a little bit. Last ping before closing for inactivity.

**Comment by r39132** [comment]
@msumit 

screenshot 2016-10-17 12 50 09

Hmn.. deletion is still possible.

**Comment by msumit** [comment]
@r39132 I guess `delete` is good to have.. in case someone wants to do a cleanup of a bad run?

**Comment by r39132** [comment]
Where are you on this PR? Starting my 3 day counter :-)

**Comment by msumit** [comment]
@r39132 everything is done.. except travis showing some false failures..

**Comment by r39132** [comment]
+1

**Pull request closed without merging**